mitodl/mitxonline
# Requests additional information from the learner when they register.

#### Pre-Flight checklist


#### What are the relevant tickets?

Closes #1494

#### What's this PR do?

Updates the registration workflow to include a step for the learner to provide additional demographic information about themselves. This is the same info that we request when they navigate to a course for the first time and is optional during the registration flow. 

This also makes a small change to make the normal profile form set the additional detail flag when saving there (otherwise learners might get prompted to supply data that they've already supplied). 

#### How should this be manually tested?

As this flow involves page transitions, you'll need to set the `FEATURE_ENABLE_ADDL_PROFILE_FIELDS` flag in your `.env` file to test - using the URL flag won't be sufficient.

1. Register a new account. 
2. After confirming your email, fill out the new user form.
3. You should be see a request for more information.
   1. Submitting the form without supplying data should work. In this case, the additional field flag in the user's profile should be clear.
   2. Submitting the form with data should also work. In this case, the additional field flag should be set.
   3. In both cases, you should be redirected to the dashboard.
4. If you _did not_ supply demo data, you should see the modal popup requesting demo data when you try to navigate into a course.  

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/945611/226735710-03a1ebb0-30e2-422e-af85-8de50cd33201.png)
![image](https://user-images.githubusercontent.com/945611/226735747-6c3077fd-f2e0-495c-a598-d6efe480484a.png)
